src/main/java/org/gridsuite/explore/server/services/ExploreService.java

```diff
@@ -143,42 +152,36 @@ public ExploreService(
-                contingencyListService.deleteContingencyList(elementAttributes.getElementUuid()).subscribe();
-            } else if (elementAttributes.getType().equals(FILTER)) {
-                filterService.deleteFilter(elementAttributes.getElementUuid()).subscribe();
+            IDirectoryElementsService service = genericServices.get(elementAttributes.getType());
+            if (service == null) {
+                return Mono.error(new ExploreException(UNKNOWN_ELEMENT_TYPE, "Unknown element type :" + elementAttributes.getType()));
             }
-            return directoryService.deleteElement(id, userId);
+            return service.delete(id, userId).doOnSuccess(e -> directoryService.deleteElement(id, userId).subscribe());
         });
     }
 
-    public Mono<List<ElementAttributes>> getElementsMetadata(List<UUID> ids) {
+    public Flux<ElementAttributes> getElementsMetadata(List<UUID> ids) {
         Mono<Map<UUID, ElementAttributes>> elementsAttributesMono = directoryService.getElementsAttribute(ids).collect(Collectors.toMap(ElementAttributes::getElementUuid, Function.identity()));
+        return elementsAttributesMono.flatMapMany(elementsAttributes -> {
+            Map<String, List<UUID>> uuidsByType = elementsAttributes.values().stream().collect(groupingBy(ElementAttributes::getType,
+                Collectors.mapping(ElementAttributes::getElementUuid, Collectors.toList())));
 
-        return elementsAttributesMono.flatMap(elementsAttributes -> {
-            List<UUID> filtersUuids = elementsAttributes.values().stream().filter(elementAttributes -> elementAttributes.getType().equals(FILTER)).map(ElementAttributes::getElementUuid).collect(Collectors.toList());
-            List<UUID> contingencyListsUuids = elementsAttributes.values().stream().filter(elementAttributes -> elementAttributes.getType().equals(CONTINGENCY_LIST)).map(ElementAttributes::getElementUuid).collect(Collectors.toList());
-
-            Mono<List<Map<String, Object>>> filtersMetadataMono = filterService.getFilterMetadata(filtersUuids).collectList();
-            Mono<List<Map<String, Object>>> contingencyListMetadataMono = contingencyListService.getContingencyListMetadata(contingencyListsUuids).collectList();
-
-            Mono<Tuple2<List<Map<String, Object>>, List<Map<String, Object>>>> metadata = Mono.zip(filtersMetadataMono, contingencyListMetadataMono);
-
-            return metadata.map(data -> {
-                Map<String, Map<String, Object>> filtersMetadataMap = data.getT1().stream().collect(Collectors.toMap(e -> e.get("id").toString(), Function.identity()));
-                Map<String, Map<String, Object>> contingenciesMetadataMap = data.getT2().stream().collect(Collectors.toMap(e -> e.get("id").toString(), Function.identity()));
+            if (!genericServices.keySet().containsAll(uuidsByType.keySet())) {
+                return Flux.error(() -> new ExploreException(UNKNOWN_ELEMENT_TYPE, "Unknown element type "));
+            }
```

Comment:
Move  the unknow type exception creation in ExploreException class ?
`    public static ExploreException createEquipmentTypeUnknown(String type) {
        Objects.requireNonNull(type);
        return new ExploreException(Type.UNKNOWN_ELEMENT_TYPE, "The element type : " + type + " is unknown");
    }`

Reply:
done